nav2_map_server/include/nav2_map_server/vector_object_shapes.hpp

```diff
@@ -0,0 +1,426 @@
+   */
+  bool isPointInside(const double px, const double py) const;
+
+  /**
+   * @brief Puts shape borders on map.
+   * @param map Output map pointer
+   * @param overlay_type Overlay type
+   */
+  void putBorders(nav_msgs::msg::OccupancyGrid::SharedPtr map, const OverlayType overlay_type);
+
+  /**
+   * @brief Gets Circle parameters
+   * @return Circle parameters
+   */
+  nav2_msgs::msg::CircleObject::SharedPtr getParams() const;
+
+  /**
+   * @brief Tries to update Circle parameters
+   * @return False in case of inconsistent shape
+   */
+  bool setParams(const nav2_msgs::msg::CircleObject::SharedPtr params);
+
+  /**
+   * @brief Gets the value of the shape.
+   * @return OccupancyGrid value of the shape
+   */
+  int8_t getValue() const;
+
+  /**
+   * @brief Gets frame ID of the shape.
+   * @return Frame ID where the shape is
+   */
+  std::string getFrameID() const;
+
+  /**
+   * @brief Gets UUID of the shape.
+   * @return Shape UUID string
+   */
+  std::string getUUID() const;
+
+  /**
+   * @brief Whether the shape to be filled or only its borders to be put on map.
+   * @return True if shape to be filled
+   */
+  bool isFill() const;
+
+  /**
+   * @brief Transforms shape coordinates to a new frame.
+   * @param to_frame Frame ID to transform to
+   * @param tf_buffer TF buffer to use for the transformation
+   * @param transform_tolerance Transform tolerance
+   * @return Whether it was transformed successfully
+   */
+  bool toFrame(
+    const std::string & to_frame,
+    const std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+    const double transform_tolerance);
+
+protected:
+  /**
+   * @brief Checks that shape is consistent for further operation
+   * @return False in case of inconsistent shape
+   */
+  bool checkConsistency();
+
+  /**
+   * @brief Converts circle center to map coordinates
+   * considering FP-accuracy loosing on small values when using conventional
+   * nav2_util::worldToMap() approach
+   * @param map Map pointer
+   * @param mcx Output X-coordinate of associated circle center on map
+   * @param mcy Output Y-coordinate of associated circle center on map
+   * @return True if the conversion was successful
+   */
+  bool centerToMap(
+    nav_msgs::msg::OccupancyGrid::ConstSharedPtr map,
+    unsigned int & mcx, unsigned int & mcy);
+
+  /**
+   * @brief Put Circle's point on map
+   * @param mx X-coordinate of given point
+   * @param my Y-coordinate of given point
+   * @param map Output map pointer
+   * @param overlay_type Overlay type
+   */
+  void putPoint(
+    unsigned int mx, unsigned int my,
+    nav_msgs::msg::OccupancyGrid::SharedPtr map,
+    const OverlayType overlay_type);
+
+  /// @brief Input circle parameters (could be in any frame)
+  nav2_msgs::msg::CircleObject::SharedPtr params_;
+  /// @brief Circle center in the map's frame
+  geometry_msgs::msg::Point32::SharedPtr center_;
+};
+
+/// @brief Functor class used in raytraceLine algorithm
+class MapAction
```

Comment:
In vector object shapes file?

Reply:
Isn't this more a util?

Reply:
Moved to `vector_object_utils.hpp` in https://github.com/ros-planning/navigation2/pull/3930/commits/43af6df733237b6f5ef8c91d0757b2ceda43a2de